### PR TITLE
Add support for collecting Track 2 Azure SDK client activities

### PR DESCRIPTION
**Pull request by pakrym**
cc @lmolkova @SergeyKanzhelev

### CODE REVIEW

File: WEB/Src/DependencyCollector/Shared.Tests/Implementation/AzureSdkDiagnosticListenerTest.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                Assert.IsTrue(telemetry.Properties.TryGetValue("tracestate", out var tracestate));
+                Assert.AreEqual("state=some", tracestate);
+            }
+        }
+
+        [TestMethod]
+        public void AzureClientSpansAreMarkedAsFailed()
+        {
+            using (var listener = new DiagnosticListener("Azure.SomeClient"))
+            using (var module = new DependencyTrackingTelemetryModule())
```
Comment (cijothomas) [inline WEB/Src/DependencyCollector/Shared.Tests/Implementation/AzureSdkDiagnosticListenerTest.cs]: Can you add one test with EnableAzureSdkTelemetryListener = false?

Comment (pakrym) [reply WEB/Src/DependencyCollector/Shared.Tests/Implementation/AzureSdkDiagnosticListenerTest.cs]: Done

**Comment by pakrym** [comment]
Output from my sample app:

image

**Comment by lmolkova** [comment]
Looking into the screenshot, there are several things we need to fix:

1. HTTP requests - we need to dedup double-instrumentation for HTTP from azure SDK and in AppInsights. In existing listeners (for core and desktop), we should back-off if traceparent header is present.

2. Dependency types: OTHER - we'll need to fix them.   
  - EventHub processor should be request-telemetry (not dependency) All server span should translate  into requests
  - Internal spans (convenience layer) should be dependency telemetry with kind InProc
  - Client spans should translate into Dependency with the appropriate type. EventHubs is the only SDK that emits such spans now and properties on the dependency should match what we report for existing EventHub (Track1) SDK in the existing eventhub listener. It includes dependency type,  data and target.

I can help with this.

**Comment by pakrym** [comment]
> I can help with this.

I would appreciate the help.

**Comment by lmolkova** [comment]
@pakrym can you please also update changelog

**Comment by pakrym** [comment]
> @pakrym can you please also update changelog

Done

**Comment by lmolkova** [comment]
@pakrym please also merge https://github.com/pakrym/ApplicationInsights-dotnet/pull/2 - I broke some tests....

**Comment by pakrym** [comment]
> @pakrym please also merge pakrym#2 - I broke some tests....

Done

**Comment by lmolkova** [comment]
last call for comments. Going to merge in 2 hours.

**Pull request merged**